crates/biome_js_analyze/src/lib.rs

```diff
@@ -39,21 +40,42 @@ pub static METADATA: LazyLock<MetadataRegistry> = LazyLock::new(|| {
     metadata
 });
 
+#[derive(Default)]
+pub struct JsAnalyzerServices {
+    dependency_graph: Arc<DependencyGraph>,
+    project_layout: Arc<ProjectLayout>,
+    source_type: JsFileSource,
+}
+
+impl From<(Arc<DependencyGraph>, Arc<ProjectLayout>, JsFileSource)> for JsAnalyzerServices {
+    fn from(
+        (dependency_graph, project_layout, source_type): (
+            Arc<DependencyGraph>,
+            Arc<ProjectLayout>,
+            JsFileSource,
+        ),
+    ) -> Self {
+        Self {
+            dependency_graph,
+            project_layout,
+            source_type,
+        }
+    }
+}
+
 /// Run the analyzer on the provided `root`: this process will use the given `filter`
 /// to selectively restrict analysis to specific rules / a specific source range,
 /// then call `emit_signal` when an analysis rule emits a diagnostic or action.
 /// Additionally, this function takes a `inspect_matcher` function that can be
 /// used to inspect the "query matches" emitted by the analyzer before they are
 /// processed by the lint rules registry
-#[expect(clippy::too_many_arguments)]
 pub fn analyze_with_inspect_matcher<'a, V, F, B>(
     root: &LanguageRoot<JsLanguage>,
     filter: AnalysisFilter,
     inspect_matcher: V,
     options: &'a AnalyzerOptions,
     plugins: Vec<Box<dyn AnalyzerPlugin>>,
-    source_type: JsFileSource,
-    project_layout: Arc<ProjectLayout>,
+    services: JsAnalyzerServices,
```

Comment:
I believe it's now the moment to create a struct to pass all these arguments, what do you think? It would be a struct and a callback

Reply:
I just added the `JsAnalyzerServices` struct to avoid adding too many arguments, while not necessarily needing to put the rest of them in a struct :sweat_smile: